### PR TITLE
Add monitoring of entities scoped under a fileview for the last X days

### PR DESCRIPTION
**Pull request by thomasyu888**
* Creates `synapsemonitor monitor` cli command
* Monitors a single Synapse project/entities scoped in a fileview for changes.  Will notify users for any files that were added or updated in the past X number of days.

Motivate users to use file views if there is a desire to track files from many projects.

### CODE REVIEW

**Review by jaybee84 — COMMENTED** [review]
Two updates that we discussed today which would be helpful to add:
1. Writing the output table of contents directly to console or redirected to a file
2. Adding the option of adding a fileview_id instead of project_id

And one wishlist item :) 
Option to add multiple fileview_ids/project_ids so that the email sent out  or output is essentially a digest of updates to all fileviews that I may be monitoring

**Comment by thomasyu888** [comment]
Thanks @jaybee84 for the review. I made revisions for new usage:

```
synapsemonitor project_or_view {project or view} -d 50 --output output.csv
```

**Comment by thomasyu888** [comment]
@jaybee84 brought up a really point that currently the behavior of the command when specifying a project is that a file view named `(monitor) file view)` is automatically created.  This could be redundant if a project already has a file view.  The main questions are: 
* Should users be able to specify a project id?
* Should the file view `(monitor) file view` be automatically created?

Below are some proposed changes:

1. **Add documentation:**  If the user provides a fileview synapse id then no file view is created, if the user provides a project id, it assumes the project has no file view, and will create the file view `(monitor) file view`.  Synapse is smart enough to not create this file view twice as it will pull the existing `(monitor) file view` that already exists in a project if the code has already been run once.
2. **Only allow for file views**:  If the user can only specify a file view synapse id, then it would be up to the user to create a file view for a project / projects if the view doesn't already exist.  A cli to create a file view could definitely be added
3. **New Synapse Feature:** If a default file view was created per project (not containing all annotations), my code could just pull from that.  The difficulty will be when you want to create a view with multiple projects.  You will be duplicating some of the code no matter what.

File: synapsemonitor/monitor.py [hunk 1]
```diff
@@ -1,109 +1,155 @@
...
+
+    Args:
+        syn: Synapse connection
+        view_id: Synapse View Id
+        epochtime: Epoch time in milliseconds
+
+    Returns:
+        Dataframe with updated entities
+    """
+    query = ("select id, name, currentVersion, modifiedOn, modifiedBy, "
```
Comment (jaybee84) [inline synapsemonitor/monitor.py:41]:
```suggestion
    query = ("select id, name, currentVersion, modifiedOn, modifiedBy, createdOn"
```

Comment (jaybee84) [reply synapsemonitor/monitor.py:41]: One use case is to find only newly added files

**Comment by kelshmo** [comment]
To make this code permissive to the most use cases (as discussed during Code Review meeting), we suggest taking a file view synId as the input parameter so the user can determine 1. which project to store the file view 2. what context to include in the file view (e.g. 2 folders, 2 projects, or 1 single project as you have proposed above etc.).

**Comment by jaybee84** [comment]
@marcomarasca mentioned that the platform team is having a discussion that is in line with the thoughts behind this PR. @thomasyu888 please check out this [JIRA](https://sagebionetworks.jira.com/browse/PLFM-6567)

**Comment by jaybee84** [comment]
Few thoughts from the code review including and in addition to the ones mentioned by @kelshmo earlier:
1. Mandate a `fileviewId` as input for monitor.py. Separate out `create_fileviewId` and add documentation to point to that function if user does not have `fileviewId` already created.
2. To speed up the function: check for count diff first without fetching everything. If count changes then pull details.
3. Option to create a new `fileviewId` in a **different project** to give user the capability of tracking all projects in a separate independent space. 
4. Two types of use cases seem to exist in Sage: a) fileviews with global scope for a specific project that resides in the project, b) fileviews with scopes of 1 folder or multiple folders from one project or multiple folders from multiple projects that reside in a separate project space tracked by Sage personnel
5. Re: Synapse feature for creating fileview as default of project creation: Synapse team has reservations about this but there is an ongoing discussion in the JIRA mentioned above. It was suggested that we should talk to Stacey to improve Synapse docs so that the Synapse user known that they can make a fileview to track their own files in the project that they created.

**Comment by thomasyu888** [comment]
Thanks all!  I will take a look and implement these :)

**Comment by thomasyu888** [comment]
To make things simple, I am also going to remove the `lastAuditTimeStamp` feature.  This is because, an individual needs to have write access to the fileview to add the`lastAuditTimeStamp`.  I will keep this feature on a separate PR for now: https://github.com/Sage-Bionetworks/synapseMonitor/pull/10

**Review by mattfazza — APPROVED** [review]
This PR is a bundle with the other one (:

File: synapsemonitor/monitor.py [hunk 2]
```diff
@@ -1,109 +1,152 @@
...
+        Dataframe with files modified within last N days
+    """
+
+    entity = syn.get(view_id)
+    # Code review decision to only allow file views so that
+    # Users can decide where they want to store their own fileview
+    # and can choose the scope of the fileview. (Scope meaning)
+    # the entities they want to have tracked.
+    if not isinstance(entity, synapseclient.EntityViewSchema):
+        raise ValueError(f"{view_id} must be a Synapse File View")
```
Comment (jaybee84) [inline synapsemonitor/monitor.py:134]:
```suggestion
        raise ValueError(f"{view_id} must be a Synapse File View. Please review 'create_file_view' function to create a Synapse File View")
```

Comment (jaybee84) [reply synapsemonitor/monitor.py:134]: Just adding some user-friendly error message

**Review by jaybee84 — APPROVED** [review]

**Comment by thomasyu888** [comment]
Thanks for the reviews @mattfazza  and @jaybee84

**Pull request merged**